nixos/modules/services/misc/ethminer.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ethminer;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.ethminer = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "enable ethminer ether mining";
+      };
+
+      recheck = mkOption {
+        type = types.string;
+        default = "1000";
+        description = "time to recheck";
+      };
+
+      toolkit = mkOption {
+        type = types.string;
+        default = "cuda";
+        description = "cuda or opencl";
+      };
+
+      apiPort = mkOption {
+        type = types.string;
+        default = "-3333";
+        description = "ethminer api port";
+      };
+
+      wallet = mkOption {
+        type = types.string;
+        default = "0xa7E7030dbFf0Be4695a36C413C299Fd3469E83f3";
+        description = "ethereum wallet address";
+      };
+
+      pool = mkOption {
+        type = types.string;
+        default = "eth-us-east1.nanopool.org";
+        description = "mining pool address";
+      };
+
+      stratumPort = mkOption {
+        type = types.string;
+        default = "9999";
+        description = "stratum protocol tcp port";
+      };
+
+      rig = mkOption {
+        type = types.string;
+        default = "mining-rig-name";
+        description = "mining rig name";
+      };
+
+      notify = mkOption {
+        type = types.string;
+        default = "email%40example.org";
```

Comment:
when there is no sensible default, maybe specify an example instead of a default.
Same for the wallet, I guess.